client/odysseus/message/index.tsx

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+import { RefObject, useState } from 'react';
+import AsyncLoad from 'calypso/components/async-load';
+import type { Message } from '../types';
+
+import './style.scss';
+
+type ChatMessageProps = {
+	message: Message;
+	isLast: boolean;
+	messageEndRef: RefObject< HTMLDivElement >;
+};
+
+const ChatMessage = ( { message, isLast, messageEndRef }: ChatMessageProps ) => {
+	const [ liked, setLiked ] = useState( false );
+	const [ disliked, setDisliked ] = useState( false );
+
+	const isUser = message.role === 'user';
+
+	const onLike = () => {
+		if ( liked ) {
+			setLiked( false );
+			setDisliked( false );
+			return;
+		}
+		setLiked( true );
+		setDisliked( false );
+	};
+
+	const onDislike = () => {
+		if ( disliked ) {
+			setLiked( false );
+			setDisliked( false );
+			return;
+		}
+		setLiked( false );
+		setDisliked( true );
+	};
```

Comment:
Assuming that `like` and `dislike` are counter actions, can't we just have a single `liked / setLiked` state and use that for both `like` and `dislike` actions? Otherwise, maintaining 2 separate states that are always opposite feels like a code smell and can lead to unexpected bugs if we fail to properly always maintain the opposite state?